Lib/test/test_logging.py

```diff
@@ -4432,12 +4432,17 @@ def test_dict_arg(self):
     def _extract_logrecord_process_name(key, logMultiprocessing, conn=None):
         prev_logMultiprocessing = logging.logMultiprocessing
         logging.logMultiprocessing = logMultiprocessing
+        old_mp = None
         try:
             import multiprocessing as mp
             name = mp.current_process().name
 
             r1 = logging.makeLogRecord({'msg': f'msg1_{key}'})
+
+            old_mp = sys.modules['multiprocessing']
             del sys.modules['multiprocessing']
+            assert old_mp
+
             r2 = logging.makeLogRecord({'msg': f'msg2_{key}'})
```

Comment:
Maybe `with support.swap_item(sys.modules, 'multiprocessing', None):` can be used on the r2=makeLogRecord(...) call, to limit the scope of the change?

Reply:
Yes, this works! Thanks for the suggestion.